fix: improve trap focus and release focus

xiaoxli opened the pull request:
To solve this bug: https://watchmixer.visualstudio.com/Mixer/_workitems/edit/23655/

Add an optional parameter to release focus, to allow not focusing on previous state when release focus

connor4312 commented:
I'm a little confused. The bug sounds like scroll logic not triggering / properly when focus is restored to the original node. Can you add some details around how this PR fixes the bug?

xiaoxli commented:
https://watchmixer.visualstudio.com/Mixer/_git/5018684a-2c8f-48bf-9e45-944a7c18e146?path=%2Fpackages%2Fplatforms%2Ffrontend2%2Fsrc%2Fbundles%2Fxbox%2Fxbox%2Fbrowse%2Fsearch%2Fsearch-page.component.html&version=GBxiaoxli%2Ffrontend2%2Ffix%2Fsearch-page-focus&line=10&lineStyle=plain&lineEnd=11&lineStartColumn=1&lineEndColumn=1
https://watchmixer.visualstudio.com/Mixer/_git/5018684a-2c8f-48bf-9e45-944a7c18e146?path=%2Fpackages%2Fplatforms%2Ffrontend2%2Fsrc%2Fbundles%2Fxbox%2Fxbox%2Fbrowse%2Fsearch%2Fsearch-page.component.html&version=GBxiaoxli%2Ffrontend2%2Ffix%2Fsearch-page-focus&line=14&lineStyle=plain&lineEnd=15&lineStartColumn=1&lineEndColumn=1
The reason of this bug is someone added the releasefocus when the focus is in search bar, and trap focus when focus is in search result. 
1. The first problem is when switch focus in search results, search result will be added to the focusStack infinitely. So I added the check in trapFocus to prevent it. 
2. When focus is in the search bar, it releases the focus, and it will cause the focus to go back to the previous focus root's focused element, then it will always jump back to the first card of the game row. So the focus is stuck, and never go to the search bar. So here I added an optional parameter to prevent this.

I don't know if there is a better way to fix it. This is a experimental fix. Please let me know if you have better ideas @connor4312 
Also this is the frontend2 change: https://watchmixer.visualstudio.com/Mixer/_git/core-frontend/pullrequest/3527?_a=files

xiaoxli commented:
Closing because using another fix
